server/grpc/grpc.go

```diff
@@ -794,7 +794,9 @@ func (g *grpcServer) Deregister() error {
 	if logger.V(logger.InfoLevel, logger.DefaultLogger) {
 		logger.Infof("Deregistering node: %s", node.Id)
 	}
-	if err := config.Registry.Deregister(service); err != nil {
+
+	opts := []registry.DeregisterOption{registry.DeregisterDomain(g.opts.Namespace)}
```

Comment:
probably don't need to construct the slice here only to deconstruct it in the next line